plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/fs/lock/FileLockController.java

```diff
@@ -0,0 +1,251 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2025 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.model.fs.lock;
+
+import com.google.gson.Gson;
+import org.jkiss.code.NotNull;
+import org.jkiss.code.Nullable;
+import org.jkiss.dbeaver.DBException;
+import org.jkiss.dbeaver.Log;
+import org.jkiss.dbeaver.utils.GeneralUtils;
+import org.jkiss.utils.IOUtils;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.UUID;
+
+/**
+ * File based resource locks
+ */
+public class FileLockController {
```

Comment:
What does it even do? I see that it creates files with random names somewhere inside the workspace folder. It's now how lock files work. They prevent a directory from being deleted. Moreover, it doesn't even work, e.g., on Windows.
I believe it serves a different purpose, but I don't quite see it. Is it supposed to be used across multiple services?

Reply:
it creates a special lock in the workspace, and as long as this file exists (or the timeout is not exceeded) - other processes/requests to the file will wait for the lock to be released (only if the file was accessed via the lock controller - direct io operations are still available)
this is independent of the operating system, as it is custom logic, and is not related to nio FileLock